packages/block-library/src/cover/editor.scss

```diff
@@ -118,3 +117,10 @@
 .color-block-support-panel__inner-wrapper > :not(.block-editor-tools-panel-color-gradient-settings__item) {
 	margin-top: $grid-unit-30;
 }
+
+// Backward compatibility for v14 deprecation.
+.wp-block-cover:not(.has-no-zindex-in-inner-blocks) {	// Shown while media is being uploaded
+	.components-spinner {
+		z-index: z-index(".wp-block-cover__inner-container");
+	}
+}
```

Comment:
> Do we need to scope this by the proposed CSS class?

We could keep it as it was. I changed it more as part of the z-index cleanup.
I don't have a strong opinion about this element, so if you have, let me know that I keep this z-index.

> At a quick glance, the positioning of the spinner before the inner container is the same on trunk and this PR. In both cases the spinner is rendered beneath the inner container.

Just notice that if we keep the z-index in the spinner and remove from the inner blocks (where it's important to fix the issue), it would change the original behavior that you described. I didn't test, but I think it would probably be better because I suspect we would always want the spinner in the front.

Reply:
I'm not following that last suggestion sorry. Could you clarify it for me?

If you're suggesting that we change the behaviour of the spinner to bring it to the top, I'm not so sure about that. I'd vote that we don't introduce more changes here that aren't required for the bug fix. We should keep trunk's behaviour then make an informed decision on bringing the spinner forward, probably with some design feedback too.

Reply:
> Could you clarify it for me?

Thinking more now, that `.wp-block-cover:not(.has-no-zindex-in-inner-blocks)` probably wouldn't be needed because it is in the editor side and it would always be with the latest HTML. But we needed to remove the z-index from the spinner because the inner container doesn't have the z-index anymore, so the spinner would be above it (I suppose it's expected to have the inner container above, like in the `trunk`, so the user can edit the content while the media is being uploaded).

Here are 2 screenshots just for more clarification:

<img width="74" alt="Screenshot 2024-10-23 at 10 39 09" src="https://github.com/user-attachments/assets/9e9afc0e-ae51-4891-b0d4-76055d0a4c09">

<img width="77" alt="Screenshot 2024-10-23 at 10 16 15" src="https://github.com/user-attachments/assets/d40ae026-b50f-45ca-b4eb-c22ded36bea5">

That said, I tweaked it after applying the [new discussed approach](https://github.com/WordPress/gutenberg/pull/66249#discussion_r1812826334): https://github.com/WordPress/gutenberg/pull/66249/commits/e29b238d67fa59af474695e13f3dfb78c52d1a39

We needed to add a z-index in the inner container during the transient state (uploading) because of the [transient overlay](https://github.com/WordPress/gutenberg/commit/e29b238d67fa59af474695e13f3dfb78c52d1a39#diff-155f0f6f698abb06228620a990f9efb18dceb8f1abf883a61c4c47ecbcf1cdbfR38). It means that we would have the original issue in this moment, where the cover would appear over the navigation overlay, but I suspect it's fine since it's just this quick moment and it's only in the editor side. WDYT? If you consider it bad, we could try to explore other approaches to change the transient overlay, but I don't think it's worth it.

Reply:
I think that could be something we can iterate on in a follow-up.

Reply:
Makes sense! I suspect it would be a very low priority (I doubt a user would reproduce it 🤭), so I can open an issue after we merge this. 😉